### PR TITLE
pd: better mod structure

### PR DESCRIPTION
**Pull request by overvenus**
Moving some functions to `util` for a better mod structure.

PTAL @siddontang @BusyJay

### CODE REVIEW

File: src/pd/async/client.rs [hunk 1]
```diff
@@ -255,7 +80,9 @@ impl PdClient for RpcClient {
         req.set_store(stores);
         req.set_region(region);
 
-        let resp = try!(do_request(self, |client| client.Bootstrap(req.clone())));
+        let resp = try!(sync_request(&self.leader_client,
```
Comment (siddontang) [inline src/pd/async/client.rs]: remove sync, it is not synchronous request.

**Comment by overvenus** [comment]
`sync_request` waits request futures, in a caller's perspective, it does a synchronous request, so I add a prefix `sync`.

File: src/pd/async/util.rs [hunk 2]
```diff
@@ -171,3 +185,153 @@ impl<Req, Resp, F> Request<Req, Resp, F>
             .boxed()
     }
 }
+
+/// Do a request in synchronized fashion.
+pub fn sync_request<F, R>(client: &LeaderClient, f: F, retry: usize) -> Result<R>
```
Comment (overvenus) [inline src/pd/async/util.rs]: This function waits request futures, in a caller's perspective, it does a synchronous request, so I add a prefix `sync`.

File: src/pd/async/client.rs [hunk 3]
```diff
@@ -12,29 +12,17 @@
...
-
-use util::HandyRwLock;
+use kvproto::pdpb::{self, Member};
+use kvproto::pdpb_grpc::{PDAsync, PDAsyncClient};
 
 use super::super::PdFuture;
 use super::super::{Result, Error, PdClient};
-use super::util::LeaderClient;
-use super::super:: metrics::*;
+use super::util::*;
```
Comment (andelf) [inline src/pd/async/client.rs]: replace `*` with names

File: src/pd/async/client.rs [hunk 4]
```diff
@@ -255,7 +80,9 @@ impl PdClient for RpcClient {
         req.set_store(stores);
         req.set_region(region);
 
-        let resp = try!(do_request(self, |client| client.Bootstrap(req.clone())));
+        let resp = try!(sync_request(&self.leader_client,
+                                     |client| client.Bootstrap(req.clone()),
+                                     LEADER_CHANGE_RETRY));
         try!(check_resp_header(resp.get_header()));
```
Comment (andelf) [inline src/pd/async/client.rs]: should the closure parameter comes last?

File: src/pd/async/util.rs [hunk 5]
```diff
@@ -16,28 +16,33 @@ use std::sync::RwLock;
...
+use kvproto::pdpb::{ResponseHeader, ErrorType, GetMembersRequest, GetMembersResponse, Member};
+use kvproto::pdpb_grpc::{PDAsyncClient, PDAsync};
 
 use util::HandyRwLock;
 
-use super::super::PdFuture;
-use super::super::Result;
-use super::client::try_connect_leader;
+use super::super::{Result, Error, PdFuture};
+use super::super:: metrics::*;
```
Comment (andelf) [inline src/pd/async/util.rs]: a space in the path

File: src/pd/async/client.rs (hunk 4 above)
Comment (overvenus) [inline src/pd/async/client.rs]: Defer to @BusyJay  https://github.com/pingcap/tikv/pull/1712#discussion_r109278371

Comment (BusyJay) [inline src/pd/async/client.rs]: I'm ok no matter whether put the closure parameter to the last. As I said in the comment, "retry parameter should be the last one. (or next to f)".

File: src/pd/async/util.rs (hunk 2 above)
Comment (siddontang) [inline src/pd/async/util.rs]: I prefer define `fn sync_request_with_retries` and let sync_request call this function with LEADER_CHANGE_RETRY internally.

**Comment by overvenus** [comment]
PTAL @andelf

**Comment by andelf** [comment]
LGTM

File: src/pd/async/util.rs [hunk 6]
```diff
@@ -171,3 +185,153 @@ impl<Req, Resp, F> Request<Req, Resp, F>
...
 }
+
+/// Do a request in synchronized fashion.
+pub fn sync_request<F, R>(client: &LeaderClient, retry: usize, f: F) -> Result<R>
+    where F: Fn(&PDAsyncClient) -> GrpcFutureSend<R>
+{
+    for _ in 0..retry {
+        let r = {
+            let timer = PD_SEND_MSG_HISTOGRAM.start_timer();
+            let r = f(&client.inner.rl().client).wait();
```
Comment (BusyJay) [inline src/pd/async/util.rs]: get the future first, don't wait for it directly.

**Pull request merged**